core/models/user.py

```diff
@@ -77,6 +76,7 @@ def clean(self):
             raise ValidationError("Only andela email addresses allowed")
 
     def save(self, *args, **kwargs):
+        self.clean()
```

Comment:
Is there any real reason to call `self.clean()` explicitly here seeing that `self.full_clean()` calls it internally?

Reply:
@kmazi  self.clean() was not being called initially, because if it were, then we would always get an error when creating a user without an Andela email. So I had to call it directly to make it work.

Reply:
Try putting a debugger when not calling it explicitly to see that it's called when `full_clean` is called.